File: haskell/problem-6/original/main.hs
square :: Integer -> Integer
square n = n * n

sumOfSquares :: [Integer] -> Integer
sumOfSquares = sum . map square

squareOfSum :: [Integer] -> Integer
squareOfSum = square . sum

answer = (squareOfSum [1..100]) - (sumOfSquares [1..100])
main = print answer

File: java/problem-6/original/Main.java
import Utilities.math.original.MathUtilities;

public class Main {
    public static void main(String[] args) {
        int[] firstHundred = MathUtilities.numbersBetween(1, 100);
        int answer = squareOfSum(firstHundred) - sumOfSquares(firstHundred);

        System.out.println(answer);
    }

    private static int squareOfSum(int[] nums) {
        int sum = 0;

        for (int num : nums) {
            sum += num;
        }

        return (sum * sum);
    }

    private static int sumOfSquares(int[] nums) {
        int sum = 0;

        for (int num : nums) {
            int square = (num * num);
            sum += square;
        }

        return sum;
    }
}
